api/compute/v1alpha1/connectorcatalog_webhook.go

```diff
@@ -0,0 +1,122 @@
+// Licensed to the Apache Software Foundation (ASF) under one
```

Comment:
dose the marker `+kubebuilder:validation:Required` on package make sense? https://book.kubebuilder.io/reference/markers/crd-validation.html

Reply:
One issue we have is that the `ConnectorDefinition` might not contain all fields to be valid. For example, it could have a `SinkClass` defined but no `SourceClass` and be valid. Another thing is that some field depends on other fields e.g. having a `SinkConfigClass` defined would only be valid if a `SinkClass` is defined too. 
That's why I made the fields optional and tried to validate them only with the logic put into the webhook. 
I'm curious to hear your thoughts on this though.